File: backend/src/main/java/com/tripwise/backend/service/interfaces/TripService.java
package com.tripwise.backend.service.interfaces;

import com.tripwise.backend.dto.TripRequest;
import com.tripwise.backend.dto.TripResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Page;

import java.time.LocalDate;
import java.util.List;

public interface TripService {

    TripResponse createTrip(String email, TripRequest request);

    List<TripResponse> getMyTrips(String email);

    Page<TripResponse> getMyTrips(
            String email,
            int page,
            int size,
            String sortBy,
            String direction,
            String search,
            String destination,
            LocalDate startDateFrom,
            LocalDate startDateTo
    );

    TripResponse getTripById(String email, Long tripId);

    TripResponse updateTrip(
            String email,
            Long tripId,
            TripRequest request);

    void deleteTrip(String email, Long tripId);

    TripResponse archiveTrip(
            String email,
            Long tripId);

    List<TripResponse> searchTrips(
            String email,
            String city);
}

File: frontend/src/services/tripService.js
import api from "./api"

export const getTrips = async ({
  page = 0,
  size = 10,
  sortBy = "startDate",
  sortDirection = "asc",
  search = "",
  destination = "",
  startDateFrom = "",
  startDateTo = "",
} = {}) => {
  const params = {
    page,
    size,
    sortBy,
    sortDirection,
  }

  if (search.trim()) {
    params.search = search.trim()
  }

  if (destination.trim()) {
    params.destination = destination.trim()
  }

  if (startDateFrom) {
    params.startDateFrom = startDateFrom
  }

  if (startDateTo) {
    params.startDateTo = startDateTo
  }

  const response = await api.get("/api/trips", {
    params,
  })

  return response.data
}

export const getTripById = async (id) => {
  const response = await api.get(`/api/trips/${id}`)
  return response.data
}

export const createTrip = async (tripData) => {
  const response = await api.post("/api/trips", tripData)
  return response.data
}

export const updateTrip = async (id, tripData) => {
  const response = await api.put(`/api/trips/${id}`, tripData)
  return response.data
}

export const deleteTrip = async (id) => {
  const response = await api.delete(`/api/trips/${id}`)
  return response
}

export const archiveTrip = async (id) => {
  const response = await api.patch(`/api/trips/${id}/archive`)
  return response.data
}
